piptools/build.py

```diff
@@ -60,12 +137,25 @@ def build_project_metadata(
     :param src_file: Project source file
     :param build_targets: A tuple of build targets to get the dependencies
                                 of (``sdist`` or ``wheel`` or ``editable``).
+    :param attempt_static_parse: Whether to attempt to statically parse the
+                                 project metadata from pyproject.toml.
+                                 Cannot be used with ``build_targets``.
     :param isolated: Whether to run invoke the backend in the current
                      environment or to create an isolated one and invoke it
                      there.
     :param quiet: Whether to suppress the output of subprocesses.
     """
 
+    if attempt_static_parse:
+        if build_targets:
```

Comment:
Let me know if you prefer to do this implicitly based on the value of `build_targets`, instead of having a separate param

Reply:
I like how you did it.